boost-maven-plugin/src/main/java/io/openliberty/boost/liberty/LibertyPackageMojo.java

```diff
@@ -131,15 +166,25 @@ private void validateSpringBootUberJAR() throws MojoExecutionException {
      * 
      * @throws MojoExecutionException
      */
-    private void copySpringBootUberJar() throws MojoExecutionException {
+    private void copySpringBootUberJar(File springBootUberJar, boolean attach) throws MojoExecutionException {
         try {
-            if (!SpringBootUtil.copySpringBootUberJar(project.getArtifact().getFile(), BoostLogger.getInstance())) {
-                File springJar = new File(
-                        SpringBootUtil.getBoostedSpringBootUberJarPath(project.getArtifact().getFile()));
-                if (net.wasdev.wlp.common.plugins.util.SpringBootUtil.isSpringBootUberJar(springJar)) {
-                    getLog().debug("Copying back Spring Boot Uber JAR as project artifact.");
-                    FileUtils.copyFile(springJar, project.getArtifact().getFile());
+            File springBootUberJarCopy = null;
+            if (springBootUberJar != null) {
+                springBootUberJarCopy = SpringBootUtil.copySpringBootUberJar(springBootUberJar,
+                        BoostLogger.getInstance());
+            }
+
+            if (springBootUberJarCopy == null) { // Copy didn't happen
+                if (attach) {
```

Comment:
What scenario causes us to fall into this path?   The copy to xxx.spring failed, but we're going to copy back from xxx.spring to the Uber JAR?    I'm not following.

Reply:
As stated in the previous comment, this is for when the user is running the build a second time without cleaning (what the invoker `mvn verify` tests test for)